libraries/AP_WheelEncoder/AP_WheelEncoder.cpp

```diff
@@ -227,12 +227,13 @@ float AP_WheelEncoder::get_wheel_radius(uint8_t instance) const
     return _wheel_radius[instance];
 }
 
-// get the total distance travelled in meters
-Vector3f AP_WheelEncoder::get_position(uint8_t instance) const
+// return a 3D vector defining the position offset of the center of the wheel in meters relative to the body frame origin
+const Vector3f &AP_WheelEncoder::get_pos_offset(uint8_t instance) const
 {
+    static Vector3f zeroVec;
     // for invalid instances return zero vector
     if (instance >= WHEELENCODER_MAX_INSTANCES) {
-        return Vector3f();
+        return zeroVec;
```

Comment:
If I follow correctly you use a static Vector3 that is zero to prevent create a zero vector. I think it is not worthly as the case where it is needed aren't numerous, no ?

Reply:
I use static vector because get_pos_offset() function returns by reference.